Source/ACE/Network/GameEvent/Events/GameEventApproachVendor.cs

```diff
@@ -17,8 +17,8 @@ public GameEventApproachVendor(Session session, WorldObject wo, List<WorldObject
             Writer.Write((uint)wo.MerchandiseMinValue);
             Writer.Write((uint)wo.MerchandiseMaxValue);
             Writer.Write((uint)1); // magic
-            Writer.Write((float)0.8); // buy_price
-            Writer.Write((float)10); // sell_price
+            Writer.Write((float)1); // buy_price
```

Comment:
PropertyDouble.BuyPrice / 37
PropertyDouble.SellPrice / 38

Use those instead, and null coalesce them to 0.8 for buy and 1.0 for sell.